Introduce instrument naming convention for used (#523)
open-telemetry/semantic-conventions: docs/general/metrics.md

for the total amount of memory on a system.

- **usage** - an instrument that measures an amount used out of a known total

> (**limit**) and has dimensions which add up to the **limit** should be called

Reviewer: ```suggestion
(**limit**) and has dimensions which add up to the **limit**. The metric should be named
```

Maybe? I feel something is missing in the text as of now.

> `entity.usage`. For example, `system.memory.usage` with attribute

Reviewer 2: ```suggestion
`<entity>.usage`. For example, `system.memory.usage` with attribute
```

So no one thinks `entity.usage` is the actual instrument name to be used 🙂

> `state = used | cached | free | ...` for the amount of memory in a each state.

Reviewer: ```suggestion
`<entity.state> = used | cached | free | ...` for the amount of memory in a each state.
```
To show the namespace in metric attributes

- **used** - an instrument that measures an amount used but does not have

> a limit (or the limit is unknowable) should be called `entity.used`.

Reviewer 2: ```suggestion
a limit (or the limit is unknowable) should be called `<entity>.used`.
```

> a limit (or the limit is unknowable) should be called `entity.used`.

Reviewer 2: unknowable (impossible to ever know) or unknown (including "didn't bother retrieving it")?

For example, the maximum possible amount of virtual memory that a process

> may consume may fluctuate over time and is not typically known.

Reviewer 2: ```suggestion
may consume may fluctuate over time and is not typically known,
so only the currently used memory is reported as `*.memory.used`.
```

for the total amount of memory on a system.

- **usage** - an instrument that measures an amount used out of a known total

> (**limit**) and has dimensions which add up to the **limit** should be called

Reviewer 3: This no longer accounts for "soft limits", like k8s limits.  I think we agreed it was ok for usage + soft limits.  Are you suggesting we remove that use case? if so we need to revisit some k8s / container metrics.

`<entity>.usage`. For example, `system.memory.usage` with attribute
`<entity.state> = used | cached | free | ...` for the amount of memory in a each state.

- **used** - an instrument that measures an amount used but does not have
a limit (or the limit is unknowable) should be called `<entity>.used`.
For example, the maximum possible amount of virtual memory that a process
may consume may fluctuate over time and is not typically known,

> so only the currently used memory is reported as `*.memory.used`.

Reviewer 3: I'm still skeptical on this.  Do you have any examples that aren't memory based here?
